partitioned-heat-conduction/solver-fenics/heat_pySDC.py

```diff
@@ -0,0 +1,372 @@
+from fenics import Function, FunctionSpace, Expression, Constant, DirichletBC, TrialFunction, TestFunction, \
+    File, solve, grad, inner, dx, interpolate, VectorFunctionSpace, MeshFunction, MPI
+from fenicsprecice import Adapter
+from errorcomputation import compute_errors
+from my_enums import ProblemType, DomainPart
+import argparse
+import numpy as np
+from problem_setup import get_geometry
+import sympy as sp
+
+from pySDC.implementations.sweeper_classes.imex_1st_order_mass import imex_1st_order_mass
+from pySDC.implementations.controller_classes.controller_nonMPI import controller_nonMPI
+from heat_pySDC_problem_class import fenics_heat_2d
+
+
+def determine_gradient(V_g, u, flux):
+    """
+    compute flux following http://hplgit.github.io/INF5620/doc/pub/fenics_tutorial1.1/tu2.html#tut-poisson-gradu
+    :param V_g: Vector function space
+    :param u: solution where gradient is to be determined
+    :param flux: returns calculated flux into this value
+    """
+
+    w = TrialFunction(V_g)
+    v = TestFunction(V_g)
+
+    a = inner(w, v) * dx
+    L = inner(grad(u), v) * dx
+    solve(a == L, flux)
+
+
+def setup_problem(
+        participant_name,
+        domain_mesh,
+        coupling_boundary,
+        remaining_boundary,
+        V,
+        u_D,
+        forcing_expr,
+        precice,
+        coupling_expression,
+        dt,
+        res_tol=1e-11,
+        maxiter=40,
+        quad_type='LOBATTO',
+        num_nodes=4,
+        logger_level=30):
+    """
+    Setup the problem for pySDC controller
+    :param participant_name: name of the participant
+    :param domain_mesh: mesh of the domain
+    :param coupling_boundary: boundary where coupling happens
+    :param remaining_boundary: remaining boundary
+    :param V: function space
+    :param u_D: initial condition
+    :param forcing_expr: forcing term
+    :param precice: preCICE adapter
+    :param coupling_expression: coupling expression
+    :param dt: pySDC time step size
+    :return: controller, problem class
+
+    pySDC can be installed via `pip install pySDC`.
+    If you want to use the developer version, the source code repository can be cloned from "https://github.com/Parallel-in-Time/pySDC".
+    For more information on pySDC, see also "https://parallel-in-time.org/pySDC/"
+    """
+
+    # initialize controller parameters
+    controller_params = {
+        'logger_level': logger_level
+    }
+
+    # fill description dictionary for easy instantiation
+    description = {
+        'problem_class': fenics_heat_2d,
+        'problem_params': {
+            'mesh': domain_mesh,
+            'function_space': V,
+            'coupling_boundary': coupling_boundary,
+            'remaining_boundary': remaining_boundary,
+            'solution_expr': u_D,
+            'forcing_term_expr': forcing_expr,
+            'precice_ref': precice,
+            'coupling_expr': coupling_expression,
+            'participant_name': participant_name
```

Comment:
```suggestion
            'participant_name': precice.get_participant_name()
```
But maybe you do not even need this?